os/pm/pm_activity.c

```diff
@@ -139,6 +143,31 @@ void pm_activity(int domain, int priority)
 		 */
 
 		now = clock_systimer();
+
+#ifdef CONFIG_PM_METRICS
+		struct pm_accumchange_s *node = NULL;
+
+		if (name == NULL) {
+			pmdbg("ERROR: name is null\n");
+		} else {
+			for (node = (struct pm_accumchange_s*)sq_peek(&(pdom->each_accum)); node; node = (struct pm_accumchange_s*)(node->entry.flink)) {
+				if (!strncmp(node->name, name, strlen(name) + 1)) {
+					uint32_t next_accum = node->accum + priority;
+					if (next_accum > UINT16_MAX) {
+						next_accum = UINT16_MAX;
+					}
+					node->accum = next_accum;
+					break;
+				}
+			}
+		}
+
+		struct pm_statechange_s *sc = (struct pm_statechange_s*)dq_tail(&(pdom->history));
+		if (pdom->state == PM_SLEEP && strncmp(sc->wakeup_reason, "None", strlen("None") + 1) == 0) {
```

Comment:
I think you'd better define `"None"` to global, because it is used many places. Then we can easily change the reason string.